src/Nethermind/Nethermind.Runner/Ethereum/Steps/InitializeBlockchainAuRa.cs

```diff
@@ -216,5 +218,25 @@ protected override HeaderValidator CreateHeaderValidator()
                     blockGasLimitContractTransitions.Keys.ToArray())
                 : base.CreateHeaderValidator();
         }
+
+        protected override IComparer<Transaction> CreateTxPoolTxComparer()
+        {
+            Address? transactionPriorityContractAddress = _api.ConfigProvider.GetConfig<IAuraConfig>()?.TransactionPriorityContractAddress;
+            if (transactionPriorityContractAddress != null)
+            {
+                ReadOnlyTxProcessorSource readOnlyTransactionProcessorSource = new ReadOnlyTxProcessorSource(_api.DbProvider, _api.BlockTree, _api.SpecProvider, _api.LogManager);
+                var txPriorityContract = new TxPriorityContract(_api.AbiEncoder, transactionPriorityContractAddress, readOnlyTransactionProcessorSource);
+                IBlockProcessor? blockProcessor = _api.MainBlockProcessor;
+                var whitelistContractDataStore = new HashSetContractDataStore<Address>(txPriorityContract.SendersWhitelist, blockProcessor);
```

Comment:
interesting - is the contract data store a store of data held in a smart contract?

Reply:
So we have:
a) contract
b) data store that auto-updates from contract either based on call or receipts
c) different type of data stores that are based on different collections for optimizations + data characteristic requirements